pkg/domains/k8s/client.go

```diff
@@ -52,8 +51,11 @@ func GetConfig() (*rest.Config, error) {
 	}
 	// TODO configurable settings for k8s client.
 	// Reduce client-side throttling for rapid results.
-	cfg.QPS = 100
-	cfg.Burst = 1000
-	cfg.Timeout = 5 * time.Second
+	if ConfigureConfig != nil {
+		ConfigureConfig(cfg)
+	}
 	return cfg, nil
 }
+
+// ConfigureConfig if non-nil adds global settings to the common rest configuration.
+var ConfigureConfig func(cfg *rest.Config)
```

Comment:
This is not really related to the authentication bits of this PR, but I dislike having a global variable for this purpose. I know this sounds like Java-EE, but I'd probably pass in a function that generates the client for me instead of calling a package-level function that relies on a global variable. This should also make testing more stable and parallelizable.

Reply:
I took this back out, it was needless complexity. Ive made auth-copy mode "always-on" for now, we can refine later. Since korrel8r as a REST service always acts on behalf of the caller, I can't currently see any reason for it to use its own credentials, and several security risks if it did.